R/derive_flag_select_msrc.R

```diff
@@ -0,0 +1,267 @@
+#' Merge an Existence Flag From Multiple Sources
+#'
+#' @description Adds a flag variable to the input dataset which indicates if
+#'   there exists at least one observation in one of the source datasets
+#'   fulfilling a certain condition.
+#'
+#' @param flag_events Flag events
+#'
+#'   A list of `flag_event()` objects is expected. For each event the condition
+#'   (`condition` field) is evaluated in the source dataset referenced by the
+#'   `dataset_name` field. If it evaluates to `TRUE` at least once, the new
+#'   variable is set to `true_value`.
+#'
+#' @param source_datasets Source datasets
+#'
+#'   A named list of datasets is expected. The `dataset_name` field of
+#'   `flag_event()` refers to the dataset provided in the list.
+#'
+#' @param new_var New variable
+#'
+#'   The specified variable is added to the input dataset.
+#'
+#' @param true_value True value
+#'
+#'   The new variable (`new_var`) is set to the specified value for all by
+#'   groups for which at least for one source (`sources`) the condition
+#'   evaluates to `TRUE`.
+#'
+#'   The values of `true_value`, `false_value`, and `missing_value` must be of
+#'   the same type.
+#'
+#' @param false_value False value
+#'
+#'   The new variable (`new_var`) is set to the specified value for all by
+#'   groups for which at all sources (`sources`) the condition evaluates to
+#'   `FALSE` or `NA`.
+#'
+#'   The values of `true_value`, `false_value`, and `missing_value` must be of
+#'   the same type.
+#'
+#' @param missing_value Values used for missing information
+#'
+#'   The new variable is set to the specified value for all by groups without
+#'   observations in any of the sources (`sources`).
+#'
+#'   The values of `true_value`, `false_value`, and `missing_value` must be of
+#'   the same type.
+#'
+#' @inheritParams derive_var_merged_exist_flag
```

Comment:
I am just noticing this but the phrasing here is really strange to me.  

This reads a little nicer to me:
The variables specified by the by_vars argument(s) are expected to be in the dataset.

![image](https://github.com/pharmaverse/admiral/assets/10111024/8d8da3c3-695d-4c10-ae3a-b19a5cfe56df)

Reply:
also looking at `by_vars` in other examples we talk about `exprs` here but make no mention of it.  You have to look at the examples to know that you need to wrap it in `exprs`

Reply:
Updated